packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -166,31 +171,77 @@ class DescopeWc extends BaseDescopeWc {
       }
     | undefined;
 
-  async loadSdkScripts() {
-    const flowConfig = await this.getFlowConfig();
-    const scripts = flowConfig.sdkScripts;
-
-    scripts?.forEach(async (script) => {
-      const module = await loadSdkScript(script.id);
-      module(
-        script.initArgs as any,
-        { baseUrl: this.baseUrl },
-        (result: string) => {
-          // update the context with the result, under the `resultKey` key
-          this.dispatchEvent(
-            new CustomEvent('components-context', {
-              detail: {
-                // we store the result with script.id prefix to avoid conflicts with other scripts results
-                // that may have the same key
-                [getScriptResultPath(script.id, script.resultKey)]: result,
-              },
-              bubbles: true,
-              composed: true,
-            }),
-          );
+  #scriptsRegistry = new Map<string, ScriptModule>();
+
+  loadSdkScripts(scripts: ClientScript[]) {
+    if (!scripts?.length) {
+      return null;
+    }
+
+    const createScriptCallback =
+      (
+        script: {
+          id: string;
+          resultKey?: string;
         },
-      );
+        resolve: (value: any) => void,
+      ) =>
+      (result: string) => {
+        this.dispatchEvent(
+          // update the context with the result, under the `resultKey` key
+          new CustomEvent('components-context', {
+            detail: {
+              // we store the result with script.id prefix to avoid conflicts with other scripts results
+              // that may have the same key
+              [getScriptResultPath(script.id, script.resultKey)]: result,
+            },
+            bubbles: true,
+            composed: true,
+          }),
+        );
+        resolve(script.id);
+      };
+    this.loggerWrapper.debug(
+      `Preparing to load scripts: ${scripts.map((s) => s.id).join(', ')}`,
+    );
+    const promises = Promise.all(
+      scripts?.map(async (script) => {
+        if (this.#scriptsRegistry.has(script.id)) {
+          this.loggerWrapper.debug('Script already loaded', script.id);
+          const m = this.#scriptsRegistry.get(script.id);
```

Comment:
I think it will be better to use meaningful name

Reply:
The scripts IDs are pretty much the same as the name we can assign them
i.e. grecqaptcha, forter, fingerprintjs...